Makefile

```diff
@@ -198,47 +197,22 @@ env:
 # usage
 #
 
+define help_category
+	printf "\n\033[1m%s\033[0m\n" $(1)
+endef
+
+define help_entry
+    printf "  \033[36m%-30s\033[0m %s\n" $(1) $(2)
+endef
+
+comma := ,
+
 .PHONY: help
 help:
-	@echo ""
-	@echo "# environment"
-	@echo ""
-	@echo "    $$ make env                  	# show makefile environment/variables"
-	@echo ""
-	@echo "# build"
-	@echo ""
-	@echo "    $$ make all                  	# build tracee-ebpf, tracee-rules & rules"
-	@echo "    $$ make bpf-core             	# build ./dist/tracee.bpf.core.o"
-	@echo "    $$ make bpf-nocore           	# build ./dist/tracee.bpf.XXX.o"
-	@echo "    $$ make tracee-ebpf          	# build ./dist/tracee-ebpf"
-	@echo "    $$ make tracee-rules         	# build ./dist/tracee-rules"
-	@echo "    $$ make rules                	# build ./dist/rules"
-	@echo ""
-	@echo "# install"
-	@echo ""
-	@echo "    $$ make install-bpf-nocore   	# install BPF no CO-RE obj into /tmp/tracee"
-	@echo "    $$ make uninstall-bpf-nocore 	# uninstall BPF no CO-RE obj from /tmp/tracee"
-	@echo ""
-	@echo "# clean"
-	@echo ""
-	@echo "    $$ make clean                	# wipe ./dist/"
-	@echo "    $$ make clean-bpf-core       	# wipe ./dist/tracee.bpf.core.o"
-	@echo "    $$ make clean-bpf-nocore     	# wipe ./dist/tracee.bpf.XXX.o"
-	@echo "    $$ make clean-tracee-ebpf    	# wipe ./dist/tracee-ebpf"
-	@echo "    $$ make clean-tracee-rules   	# wipe ./dist/tracee-rules"
-	@echo "    $$ make clean-rules          	# wipe ./dist/rules"
-	@echo ""
-	@echo "# test"
-	@echo ""
-	@echo "    $$ make test                 	# run all go & opa tests"
-	@echo "    $$ make test-tracee-ebpf     	# go test tracee-ebpf"
-	@echo "    $$ make test-tracee-rules    	# go test tracee-rules"
-	@echo "    $$ make test-rules           	# opa test (tracee-rules)"
-	@echo ""
-	@echo "# flags"
-	@echo ""
-	@echo "    $$ STATIC=1 make ...                 # build static binaries"
-	@echo "    $$ BTFHUB=1 STATIC=1 make ...        # build static binaries, embed BTF"
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-30s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```

Comment:
$(CMD_AWK) here instead (so user might be able to override de command (gawk, mawk, etc)).

Reply:
I find hard to understand/maintain a oneliner here. Is it possible for us to have something structured ? Even if with \ at the end lines (just so its easier to read).

Like:

```awk
BEGIN {
	FS = ":.*##";
	printf "\nUsage:\n  make \033[36m<target>\033[0m\n"
}

/^[a-zA-Z_0-9-]+:.*?##/ {
	printf "  \033[36m%-30s\033[0m %s\n", $1, $2
}

/^##@/ {
	printf "\n\033[1m%s\033[0m\n", substr($0, 5)
}
```